src/db/car/many.rs

```diff
@@ -0,0 +1,136 @@
+// Copyright 2019-2023 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+//! The [`ManyCar`] block store is the union of `N` read-only CAR-backed block
+//! stores and a single writable block store. Get requests are forwarded to each
+//! store (including the writable store) and the first hit is returned. Write
+//! requests are only forwarded to the writable store.
+//!
+//! A single z-frame cache is shared between all read-only stores.
+
+use super::{AnyCar, ZstdFrameCache};
+use crate::blocks::Tipset;
+use crate::db::MemoryDB;
+use anyhow::Context;
+use cid::Cid;
+use fvm_ipld_blockstore::Blockstore;
+use parking_lot::Mutex;
+use std::{io, path::PathBuf, sync::Arc};
+
+pub struct ManyCar<WriterT = MemoryDB> {
+    shared_cache: Arc<Mutex<ZstdFrameCache>>,
+    read_only: Vec<AnyCar<Box<dyn super::CarReader>>>,
+    writer: WriterT,
+}
+
+impl ManyCar {
```

Comment:
nit: [state migration tests](https://github.com/ChainSafe/forest/blob/main/src/state_migration/tests/mod.rs#L91) could use `ManyCar`. I can also do that in a separate PR

Reply:
nvm, bundle car is no longer needed once https://github.com/ChainSafe/forest/pull/3276 is merged